src/Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native/SoftwareEntitlementClient.cpp

```diff
@@ -742,12 +771,32 @@ std::unique_ptr<Entitlement> GetEntitlement(
         url += '/';
     }
 
-    Curl curl;
-    curl.Post(url + "softwareEntitlements?api-version=2017-05-01.5.0", entitlement_token, requested_entitlement);
-
-    curl.VerifyIntermediateCertificate(url);
+    for (unsigned int retry = 1; retry <= retries; ++retry)
+    {
+        try
+        {
+            return Curl::GetEntitlement(url, entitlement_token, requested_entitlement);
+        }
+        catch (const Curl::CurlException& e)
+        {
+            if (e.GetCode() == CURLE_OPERATION_TIMEDOUT)
```

Comment:
We're confident this is the only error type? Does this cover both the SSL connecting timeout as well as the Resolving timeout?

Reply:
This is error code 28, which we saw in all instances of the failure.  The text description for the failure changes, but the code is the same.